src/dmd/backend/blockopt.d

```diff
@@ -1025,12 +1045,19 @@ private void brrear()
                         /* the number of iterations.                    */
 
                         version (SCPP)
-                            enum additionalAnd = "b.Btry == bt.Btry";
+                        {
+                            static if (NTEXCEPTIONS)
+                                enum additionalAnd = "b.Btry == bt.Btry &&
+                                                  bt.Btry == bt.nthSucc(0).Btry";
+                            else
+                                enum additionalAnd = "b.Btry == bt.Btry";
+                        }
                         else static if (NTEXCEPTIONS)
                             enum additionalAnd = "b.Btry == bt.Btry &&
-                                                  bt.Btry == bt.nthSucc(0).Btry";
+                                              bt.Btry == bt.nthSucc(0).Btry";
```

Comment:
What's the motivation for this code duplication?

Reply:
The lack of `||` for version statements.

Reply:
Oh I was wondering about the fact that you duplicated the strings too, because there are a few ways to avoid this like:

```d
version(SCPP)
    enum SCPP = 1;
```